python-spec/src/somacore/data.py

```diff
@@ -65,7 +76,46 @@ def read(
     ) -> "ReadIter[pa.Table]":
         """Reads a user-defined slice of data into Arrow tables.
 
-        TODO: Further per-param documentation.
+        :param coords: for each index dimension, which rows to read.
+            Defaults to ``()``, meaning no constraint -- all IDs.
+        :param column_names: the named columns to read and return.
+            Defaults to ``None``, meaning no constraint -- all column names.
+        :param partitions: If present, specifies that this is part of
+            a partitioned read, and which part of the data to include.
+        :param result_order: the order to return results, specified as a
+            :class:`~options.ResultOrder` or its string value.
+        :param value_filter: an optional value filter to apply to the results.
+            The default of ``None`` represents no filter. Value filter syntax
+            is implementation-defined; see the documentation for a particular
+            SOMA implementation for details.
+
+        **Indexing:**
+
+        Indexing is performed on a per-column basis for each indexed column.
+        To specify dimensions:
+
+        - A sequence of coordinates is accepted, one per indexed dimension.
+        - The sequence length must be less than or equal to the number of
+          indexed dimensions.
+        - If the sequence is shorter than the number of indexed coordinates,
+          then no constraint (i.e. ``None``) is used for the remaining
+          indexed dimensions.
+        - Specifying an empty sequence (e.g. ``()``, the default) represents
+          no constraints over any dimension, returning the entire dataset.
+
+        Each dimension may be indexed as follows:
+
+        - ``None`` or ``slice(None)`` places no constraint on the dimension.
+        - Coordinates can be specified as a scalar value, a Python sequence
+          (``list``, ``tuple``, etc.), a NumPy ndarray, an Arrow array, or
+          similar objects (as defined by ``SparseDFCoords``).
+        - Slices are doubly inclusive: ``slice(2, 4)`` means ``[2, 3, 4]``,
```

Comment:
`doubly-inclusive` is a term that I do not often see in Python docs - more common is `half-open` or `closed`, to track with traditional interval/range nomenclature.

Suggested change for clarity:   "Slices are doubly inclusive (i.e., specify a closed range): ..."

Reply:
ditto for other places we use the term.

Reply:
Changed to "closed" in all locations.